### PR TITLE
Migrate vector page caches to Caffeine LoadingCache + expose stats

### PR DESCRIPTION
**Pull request by eolivelli**
## Summary

Replaces the hand-rolled `LinkedHashMap` LRU in `PageStoreReader` and the passive `get`/`put` `SharedSegmentPageCache` with Caffeine `LoadingCache` instances. Exposes Caffeine stats as per-index gauges and adds Grafana panels for them.

### Why

- The old `PageStoreReader` LRU held a `synchronized(cache)` block on every hot-path lookup — serialising concurrent searches on the same segment.
- Neither cache deduplicated concurrent misses: two search threads missing the same `pageId` each issued a full DSM read.
- `SharedSegmentPageCache` already called `.recordStats()` but never exposed the results; there was no way to answer "is the segment cache doing anything useful?".
- The per-reader cache exposed custom `AtomicLong` hit/miss counters that duplicated Caffeine built-ins.

### What changed

- **`SharedSegmentPageCache`** — now a `LoadingCache<PageKey, byte[]>`. New `PageKey` value type (tableSpaceUUID + indexUUID + pageId — no `pageType` field, since only `TYPE_VECTOR_GRAPHCHUNK` flows through this cache today). Constructor takes a `PageLoader` functional interface; `load(PageKey)` is the single entry point. New `cleanUp()` forces async eviction maintenance to run synchronously.
- **`PageStoreReader`** — per-segment cache is now `LoadingCache<Integer, byte[]>`. The cache loader delegates to the shared cache (which has its own loader), so two tiers of single-flight loading stack cleanly. The hot read path is lock-free.
- **`PersistentVectorStore`** — constructs the shared cache with a `loadGraphChunkPage` loader that wraps `dsm.readIndexPage(...)`. Exposes 18 cache-stats accessors (nine for the shared cache, nine aggregated across per-segment `PageStoreReader` caches).
- **`IndexingServiceEngine.registerIndexMetrics`** — registers 16 new per-index gauges under scopes `{ts}_{table}_{vidx}_segment_cache_*` and `{ts}_{table}_{vidx}_page_reader_cache_*`: hits, misses, evictions, load\_success, load\_failure, load\_time\_nanos\_total, entries, size\_bytes (shared only), max\_bytes (shared only).
- **Grafana dashboards**
  - `indexing-service-dashboard.json`: two new rows (*Vector Segment Cache*, *Vector Page Reader Cache*), each with hits/misses rate, hit ratio, bytes used vs max / evictions & entries, and average page-load latency (ns).
  - `vector-search-latency-dashboard.json`: new trailing row *Vector Cache Correlation* showing aggregated segment-cache hit ratio and a stack of segment-cache miss rate vs file-server client read rate.

### Unit tests — edge cases and failure scenarios covered

**New file `SharedSegmentPageCacheTest` (28 tests):**
- Happy path: single load invokes loader once; repeated load is a hit; distinct keys coexist; the three `PageKey` fields are all part of identity.
- **Concurrent miss deduplication** — 32 threads racing on the same key; loader invoked exactly once.
- Failure modes: `DataStorageManagerException` propagates unwrapped; runtime exceptions propagate as-is; loader returning `null` surfaces as exception; failed loads are not negatively cached (retry succeeds); interleaved failures and successes are tracked independently.
- Cache-disabled (`maxBytes <= 0`): every load delegates to loader, failures still surface, null results still surface.
- Invalidation: `invalidate(PageKey)` forces refetch; null/unknown keys are safe; `clear()` drops every entry.
- Byte-weighted eviction: 20 × 1 KiB into a 4 KiB cache triggers evictions; `weightedSize()` stays under budget.
- `PageKey` value semantics: `equals`/`hashCode` match; null tableSpace/index rejected via NPE; hashes spread across buckets.
- Constructor: null loader rejected; `load(null)` rejected.
- Stats accessors: track basic activity; work when disabled.
- Loader visibility: sees exactly the `PageKey` passed by the caller.

**New `PageStoreReaderTest` cases (4 added, 17 preserved — 21 total):**
- Concurrent miss dedup within a single segment (16 threads on page 0 → one DSM read).
- Stats accessors track `loadSuccess` / `loadFailure` / `loadTimeNanos`.
- Failed loads increment the `loadFailure` stat.
- Shared cache deduplicates across `PageStoreReader.Supplier` instances.
- `close()` invalidates shared-cache entries for this segment's pages.

### Follow-up (not in this PR)

The existing `rfs_client_read_*` panels in `vector-search-latency-dashboard.json` will light up once `RemoteFileDataStorageManager.multipartIndexReaderSupplier` is threaded a real `StatsLogger` instead of the `null` passed today (line 573). That's a signature change across four implementations, deferred for a separate PR.

## Test plan

- [x] `mvn -B checkstyle:check apache-rat:check spotbugs:check install -DskipTests -Pci` — BUILD SUCCESS
- [x] `mvn -pl herddb-core -Dtest='PageStoreReaderTest,SharedSegmentPageCacheTest' test` — 49 tests, 0 failures
- [x] All three Grafana dashboard JSON files validated
- [ ] End-to-end verification on k3s-local: curl the `/metrics` endpoint after a few searches and confirm `_segment_cache_hits` / `_page_reader_cache_hits` appear non-zero

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: herddb-core/src/main/java/herddb/index/vector/PageStoreReader.java [hunk 1]
```diff
@@ -337,15 +324,24 @@ private byte[] readPayload(long pageId, int expectedType)
         }
 
         int cachedPages() {
-            synchronized (cache) {
-                return cache.size();
-            }
+            // Caffeine counts may lag on pending writes; force cleanup for tests.
+            cache.cleanUp();
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/index/vector/PageStoreReader.java]: this cleanUp is expensive, do not call it in a method that is expected to be readonly

if you need this for tests, add a method to perform the clean up and call it in the tests

File: herddb-core/src/main/java/herddb/index/vector/SharedSegmentPageCache.java [hunk 2]
```diff
@@ -19,91 +19,206 @@
...
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof PageKey)) {
+                return false;
+            }
+            PageKey k = (PageKey) o;
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/index/vector/SharedSegmentPageCache.java:103]: compare hash, if the hash are different, then early exit with false
